test/easyconfigs/easyconfigs.py

```diff
@@ -1180,6 +1180,19 @@ def test_pr_python_packages(self):
                 sanity_pip_check = ec.get('sanity_pip_check') or exts_default_options.get('sanity_pip_check')
                 if not sanity_pip_check and not any(re.match(regex, ec_fn) for regex in whitelist_pip_check):
                     failing_checks.append("sanity_pip_check should be enabled in %s" % ec_fn)
+            else:
+                # Make sure the user packages in $HOME/.local/lib/python*/ are ignored when running python commands
+                # For the EasyBlocks above this is handled automatically by setting $PYTHONNOUSERSITE
+                # Detect any code or module invocation (-m or -c), `python cc` and `python <filepath>`
+                python_re = re.compile(r'\bpython (-c|-m|cc|[^ ]*\w+.py) ')
```

Comment:
Suggested in EasyBuild conf call:

* just define `$PYTHONNOUSERSITE` globally in framework;
* or introduce a template to run `python -c -s` commands

Reply:
And the other 2 too? Do we have anything lie that already?

Enforcing that template might be tricky. And we can only have it `python -s` as we need `-m, -c, cc` and maybe others which makes using the template more typing.

Reply:
What other two variables do you mean exactly?

We have similar stuff, though not for Python yet, see for example setting of `$CUDA_CACHE_DISABLE` in `easyblock.py`

Reply:
There is also `$PIP_DISABLE_PIP_VERSION_CHECK` and `$XDG_CACHE_HOME` and later `$PIP_REQUIRE_VIRTUALENV`, see https://github.com/easybuilders/easybuild-easyblocks/pull/3460/files#diff-254aa63061aafed567ef5cc25be7b809dc9a1e4e41b5de85c2e607ea95cf33cbR123

`$CUDA_CACHE_DISABLE` is set in the `prepare_step` which IIRC isn't run for `sanity-check-only`. I think we can set `$PYTHONNOUSERSITE` and `$PIP_DISABLE_PIP_VERSION_CHECK` without issues but I'm not sure where. `$PIP_REQUIRE_VIRTUALENV` would need to be set globally and after loading modules (including fake modules)